ext/civi_contribute/Civi/Api4/Action/Payment/Get.php

```diff
@@ -69,11 +70,25 @@ public function getEntityName() {
    * @var \Civi\Api4\Generic\Result $result
    */
   public function _run(\Civi\Api4\Generic\Result $result) {
+    $wheres = $this->getWhere();
+    foreach ($wheres as $whereID => $whereValues) {
+      if ($whereValues[0] === 'contribution_id') {
+        $contributionWhereOp = $whereValues[1];
+        $contributionWhereValue = $whereValues[2];
+        unset($wheres[$whereID]);
+        $this->setWhere($wheres);
```

Comment:
Shouldn't this be outside of the loop? What happens when none of the wheres are for `contribution_id`?